core/common/src/main/java/alluxio/metrics/OperationSystemCounterSet.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.metrics;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import com.sun.management.OperatingSystemMXBean;
+import com.sun.management.UnixOperatingSystemMXBean;
+
+import java.lang.management.ManagementFactory;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A set of counters for the os metric.
+ */
+public class OperationSystemCounterSet implements MetricSet {
+
+  private OperatingSystemMXBean mOsmxb = (OperatingSystemMXBean)
+          ManagementFactory.getOperatingSystemMXBean();
+  private UnixOperatingSystemMXBean mUnixb = (UnixOperatingSystemMXBean)
+          ManagementFactory.getOperatingSystemMXBean();
+
+  @Override
+  public Map<String, Metric> getMetrics() {
+    final Map<String, Metric> gauges = new HashMap<>();
+    gauges.put("os.freePhysicalMemory", (Gauge<Long>) mOsmxb::getFreePhysicalMemorySize);
```

Comment:
i am not sure how expensive these operations are. Consider using cachedGauge if necessary?

Reply:
I think totalMemory can be converted to cachedGauge, other meitrics continue to use the gauge type, do you think it is okay?

Reply:
for all the metrics that are highly likely expensive and the user can tolerate 10 minute cache time, we should go for cached gauge.

actually @yuzhu it will be better for our cache time to be configurable